python/ql/lib/semmle/python/frameworks/internal/PoorMansFunctionResolution.qll

```diff
@@ -80,7 +126,20 @@ private DataFlow::TypeTrackingNode poorMansFunctionTracker(DataFlow::TypeTracker
  * inst = MyClass()
  * print(inst.my_method)
  * ```
+ *
+ * But is able to handle simple method calls within a class, but does not take MRO into
+ * account.
+ * ```py
+ * class MyClass:
+ *     def method1(self);
+ *         pass
+ *
+ *     def method2(self);
+ *         self.method1()
+ * ```
  */
 DataFlow::Node poorMansFunctionTracker(Function func) {
   poorMansFunctionTracker(DataFlow::TypeTracker::end(), func).flowsTo(result)
+  or
+  result = getSimpleMethodReferenceWithinClass(func)
```

Comment:
Why is this not a step in the type tracking (i.e. a case in the predicate `poorMansFunctionTracker` above) instead?

Reply:
No strong argument. I just kept things simple :shrug: (and only needed to make it slightly more powerful than it already was).

My stance on this `.qll` file is that it shouldn't even exist, and should be removed as soon as we can (that is, once we have call resolution with type-tracking instead of points-to).